test/src/test/java/hudson/model/JobTest.java

```diff
@@ -286,4 +291,35 @@ private static void tryConfigDotXml(JenkinsRule.WebClient wc, int status, String
         p.renameTo("different-name");
     }
 
+    @Issue("JENKINS-30502")
+    @Test public void testRenameTrimsSpace() throws Exception {
+        tryRename("myJob", " foo", "foo");
+        tryRename("myJob", "foo ", "foo");
+    }
+
+    private void tryRename(
+            String initialName,
+            String submittedName,
+            String correctResult) throws Exception {
+        j.jenkins.setCrumbIssuer(null);
+
+        FreeStyleProject job = createJobIfNeeded(initialName);
+        WebClient wc = j.createWebClient();
+        HtmlForm form = wc.getPage(job, "configure").getFormByName("config");
+        form.getInputByName("name").setValueAttribute(submittedName);
+        HtmlPage resultPage = j.submit(form);
+
+        String urlTemplate = "/job/{0}/rename?newName={1}";
+        String expectedURL = MessageFormat.format(
+                urlTemplate, initialName, correctResult);
```

Comment:
Yes.  I tried to keep in under 80 characters as I thought I read somewhere that the Jenkins project that we were loosely following the Sun/Oracle standards (which is 80 characters[1]), but if the project policy is 120 I can update this.  Are code style guidelines for this project published anywhere?

[1] http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-136091.html